Source/ZXing.Net.Mobile.Android/ZXingSurfaceView.cs

```diff
@@ -343,38 +368,89 @@ public void OnPreviewFrame (byte[] bytes, Android.Hardware.Camera camera)
+            var focusX = (touchX / Width * 2000) - 1000;
+            var focusY = (touchY / Height * 2000) - 1000;
+
+            // Call the autofocus with our coords
+            AutoFocus ((int)focusX, (int)focusY, true);
 
             return r;
         }
 
         public void AutoFocus ()
         {
-            AutoFocus (-1, -1);
+            AutoFocus (0, 0, false);
         }
 
         public void AutoFocus (int x, int y)
         {
+            AutoFocus (x, y, true);
+        }
+
+        void AutoFocus (int x, int y, bool useCoordinates)
+        {
             if (camera != null) {
+                var cameraParams = camera.GetParameters ();
+
                 if (!autoFocusTokenCancelSrc.IsCancellationRequested) {
                     Android.Util.Log.Debug (MobileBarcodeScanner.TAG, "AutoFocus Requested");
-                    try {                         
-                        camera.AutoFocus (this); 
+
+                    // Cancel any previous requests
+                    camera.CancelAutoFocus ();
+
+                    try {
+                        // If we want to use coordinates
+                        // Also only if our camera supports Auto focus mode
+                        // Since FocusAreas only really work with FocusModeAuto set
+                        if (useCoordinates 
+                            && cameraParams.SupportedFocusModes.Contains (Camera.Parameters.FocusModeAuto)) {
+                            // Let's give the touched area a 20 x 20 minimum size rect to focus on
+                            // So we'll offset -10 from the center of the touch and then 
+                            // make a rect of 20 to give an area to focus on based on the center of the touch
+                            x = x - 10;
+                            y = y - 10;
+
+                            // Ensure we don't go over the -1000 to 1000 limit of focus area
+                            if (x >= 1000)
+                                x = 980;
+                            if (x < -1000)
+                                x = -1000;
+                            if (y >= 1000)
+                                y = 980;
+                            if (y < -1000)
+                                y = -1000;
+
+                            // Explicitly set FocusModeAuto since Focus areas only work with this setting
+                            cameraParams.FocusMode = Camera.Parameters.FocusModeAuto;
+                            // Add our focus area
+                            cameraParams.FocusAreas = new List<Camera.Area> {
+                                new Camera.Area (new Rect (x, y, x + 20, y + 20), 1000)
+                            };
+                            camera.SetParameters (cameraParams);
+                        }
+
+                        // Finally autofocus (weather we used focus areas or not)
+                        camera.AutoFocus (this);
```

Comment:
That's not a bad point, and probably better to do now in case we allow more access to change camera parameters in the future.

Right now there's really no way the camera should be able to get into a state of FocusModeFixed, but eventually that may be possible.